File: Offer/CPP/10.cpp
class Solution
{
  public:
    int rectCover(int number)
    {
        if (number <= 2)
        {
            return number;
        }
        int f1 = 1;
        int f2 = 2;
        int f3;

        for (int i = 3; i <= number; i++)
        {
            f3 = f1 + f2;
            f1 = f2;
            f2 = f3;
        }
        return f3;
    }
};

File: Offer/Python/10.py
class Solution:
    def rectCover(self, number):
        if number < 3:
            return number
        f1 = 1
        f2 = 2
        for _ in range(3, number + 1):
            f3 = f1 + f2
            f1 = f2
            f2 = f3
        return f3
